app/desktop/studio_server/provider_api.py

```diff
@@ -894,6 +975,88 @@ async def available_ollama_models() -> AvailableModels | None:
         return None
 
 
+async def available_docker_model_runner_models() -> AvailableModels | None:
+    # Try to connect to Docker Model Runner, and get the list of installed models
+    try:
+        docker_connection = await get_docker_model_runner_connection()
+        if docker_connection is None:
+            return None
+
+        docker_models = AvailableModels(
+            provider_name=provider_name_from_id(ModelProviderName.docker_model_runner),
+            provider_id=ModelProviderName.docker_model_runner,
+            models=[],
+        )
+
+        for docker_model_id in docker_connection.supported_models:
+            # Get all Kiln models that match the Docker Model Runner model ID
+            models = models_from_docker_model_runner_id(docker_model_id)
+            for model, docker_provider in models:
+                if model and docker_provider:
+                    docker_models.models.append(
+                        ModelDetails(
+                            id=model.name,
+                            name=model.friendly_name,
+                            supports_structured_output=docker_provider.supports_structured_output,
+                            supports_data_gen=docker_provider.supports_data_gen,
+                            supports_logprobs=docker_provider.supports_logprobs,
+                            suggested_for_data_gen=docker_provider.suggested_for_data_gen,
+                            suggested_for_evals=docker_provider.suggested_for_evals,
+                            uncensored=False,
+                            suggested_for_uncensored_data_gen=False,
+                            # Docker Model Runner uses OpenAI-compatible API with JSON schema support
+                            structured_output_mode=StructuredOutputMode.json_schema,
+                        )
+                    )
+        for docker_model in docker_connection.untested_models:
+            docker_models.models.append(
+                ModelDetails(
+                    id=docker_model,
+                    name=docker_model,
+                    supports_structured_output=False,
```

Comment:
I think we can hardcode supports_structured_output to true since DMR supports json_schema constrained decoding natively. That way you won't get big ugly warnings using docker for JSON tasks.